org.adempiere.base/src/org/adempiere/base/AbstractProductPricing.java

```diff
@@ -150,7 +161,7 @@ public void setRequisitionLine(I_M_RequisitionLine reqLine, String trxName) {
 	public void setRMALine(I_M_RMALine rmaLine, String trxName) {
 		m_M_Product_ID = rmaLine.getM_Product_ID();
 		if (rmaLine.getM_RMA_ID() > 0) {
-			I_M_RMA rma = rmaLine.getM_RMA();
+			I_M_RMA rma = ((MRMALine)rmaLine).getParent();
```

Comment:
the three set methods have the same issue with the I_C_OrderLine above, I guess we should just implement the same pattern here. For me, if we wants to enforce it is using M* class (MOrderLine, MProjectLine, etc), that should be enforce at the method parameter level instead of using I* parameter and will throw class cast exception if it is not an instance of M* class.